src/components/designer/taro/actions/NewAddressModal.tsx

```diff
@@ -0,0 +1,185 @@
+  const [selectedBalance, setSelectedBalance] = useState<PTARO.TaroBalance>();
+  const [taroAddress, setTaroAddress] = useState('');
+
+  //component local variables
+  const [form] = Form.useForm();
+  const thisTaroNode = network.nodes.taro.find(
+    node => node.name === nodeName,
+  ) as TarodNode;
+  const otherTaroNodes: TaroNode[] = network.nodes.taro.filter(
+    node => node.name !== nodeName,
+  );
+
+  //When polar is first opened, we need to populate the state with the lightning node data
+  useEffect(() => syncChart(network), []);
+
+  const handleSelectedBalance = (value: PTARO.TaroBalance) => {
+    setSelectedBalance(value);
+    setSelectedGenesisBootstrapInfo(value.genesisBootstrapInfo);
+    form.setFieldsValue({
+      genesisBootstrapInfo: value.genesisBootstrapInfo,
+      amount: value.type == PTARO.TARO_ASSET_TYPE.COLLECTIBLE ? 1 : 10,
+    });
+  };
+
+  //submit
+  const newAddressAsync = useAsyncCallback(async (payload: NewAddressPayload) => {
+    try {
+      const res = await getNewAddress(payload);
+      setTaroAddress(res.encoded);
+    } catch (error: any) {
+      notify({ message: l('submitError'), error });
+    }
+  });
+
+  const handleSubmit = (values: { genesisBootstrapInfo: string; amount: string }) => {
+    if (selectedGenesisBootstrapInfo) {
+      const payload: NewAddressPayload = {
+        node: thisTaroNode,
+        genesisBootstrapInfo: values.genesisBootstrapInfo,
+        amount: values.amount,
+      };
+      newAddressAsync.execute(payload);
+    }
+  };
+
+  const handleCopy = () => {
+    message.success(l('copied', { address: ellipseInner(taroAddress, 10, 10) }), 2);
+    hideNewAddress();
+  };
+
+  let cmp: ReactNode;
+  if (!taroAddress) {
+    cmp = (
+      <>
+        <Form
+          form={form}
+          layout="vertical"
+          colon={false}
+          initialValues={{
+            amount: '10',
+            node: '',
+            genesisBootstrapInfo: '',
+          }}
+          onFinish={handleSubmit}
+        >
+          <Form.Item name="genesisBootstrapInfo" label={l('genesisBootstrapInfo')}>
+            <Input.TextArea
+              rows={5}
+              placeholder={l('genesisBootstrapInfo')}
+              onChange={e => setSelectedGenesisBootstrapInfo(e.target.value)}
+            />
+          </Form.Item>
+          <Styled.Spacer />
+
+          <Styled.TaroDataSelect
+            name="node"
+            label={l('selectBalance')}
+            taroNetworkNodes={otherTaroNodes}
+            selectBalances
+            onChange={v => handleSelectedBalance(v?.valueOf() as PTARO.TaroBalance)}
+          />
+          <Styled.Spacer />
+          <Form.Item label={l('amount')} name="amount">
+            <InputNumber
+              onChange={v => setSelectedAmount(v?.valueOf()?.toString() as string)}
+              min={'1'}
+              disabled={selectedBalance?.type === PTARO.TARO_ASSET_TYPE.COLLECTIBLE}
+            />
```

Comment:
There's a few things that could be improved here. Take a look at how `<InputNumber>` is used throughout the code base and try to mirror those.

For example, here's how it's implemented in the `OpenChannelModal` component:
```ts
        <InputNumber<number>
          disabled={openChanAsync.loading}
          formatter={v => `${v}`.replace(/\B(?=(\d{3})+(?!\d))/g, ',')}
          parser={v => parseInt(`${v}`.replace(/(undefined|,*)/g, ''))}
          style={{ width: '100%' }}
          onChange={v => setSelectedSats(v as number)}
        />
```

Reply:
Done

Reply:
I seem to be using this in everything that I have done so far. What do you think of me moving this out to its own component in a refactor? I'm thinking something like components/common/form/AmountInput.tsx?